spec/safe_migration_spec.rb

```diff
@@ -43,165 +43,336 @@ def up
+                safe_add_column :foos, :bar, :text
+                remove_column :foos, :bar, :text
+              end
             end
+
+            expect do
+              migration.suppress_messages { migration.migrate(:up) }
+            end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:remove_column is NOT SAFE!/)
           end
 
-          expect do
-            migration.suppress_messages { migration.migrate(:up) }
-          end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:rename_table is NOT SAFE!/)
-        end
+          it "raises when using default add_index method" do
+            migration = Class.new(migration_klass) do
+              def up
+                safe_create_table :foos
+                safe_add_column :foos, :bar, :text
+                add_index :foos, :bar
+              end
+            end
 
-        it "raises when using default rename_column method" do
-          migration = Class.new(migration_klass) do
-            def up
-              safe_create_table :foos
-              safe_add_column :foos, :bar, :text
-              rename_column :foos, :bar, :baz
+            expect do
+              migration.suppress_messages { migration.migrate(:up) }
+            end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:add_index is NOT SAFE!/)
+          end
+
+          it "raises when using default add_foreign_key method" do
+            migration = Class.new(migration_klass) do
+              def up
+                safe_create_table :foos
+                safe_create_table :bars
+                safe_add_column :foos, :bar_fk, :integer
+                add_foreign_key :foos, :bars, :foreign_key => :bar_fk
+              end
             end
+
+            expect do
+              migration.suppress_messages { migration.migrate(:up) }
+            end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:add_foreign_key is NOT SAFE!/)
           end
 
-          expect do
-            migration.suppress_messages { migration.migrate(:up) }
-          end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:rename_column is NOT SAFE!/)
+          it "raises when using default execute method" do
+            migration = Class.new(migration_klass) do
+              def up
+                execute "SELECT current_date"
+              end
+            end
+
+            expect do
+              migration.suppress_messages { migration.migrate(:up) }
+            end.to raise_error(PgHaMigrations::UnsafeMigrationError, /:execute is NOT SAFE!/)
+          end
         end
 
-        it "raises when using default change_column method" do
-          migration = Class.new(migration_klass) do
-            def up
-              safe_create_table :foos
-              safe_add_column :foos, :bar, :text
-              change_column :foos, :bar, :string
+        context "when not configured to disable default migration methods" do
+          before do
+            allow(PgHaMigrations.config).
```

Comment:
Our house style is to using leading rather than trailing dots; could you update?